### DISCUSSION TITLE
How to get all users which associate with roles

### DISCUSSION THREAD

**Discussion by jacksfero** [spatie/laravel-permission#2347] [Q&A] [answered]
i want to list of the all user which assigned any roles.

**Comment by erikn69** [comment, marked as answer]
Using scopes
https://spatie.be/docs/laravel-permission/v5/basic-usage/basic-usage

**Reply by jacksfero** [reply]
Hi Ernikn,
i want to get all users which have any role, i am using below, but not getting
$users = User::role('writer')->get();
and not able to pass blank
$users = User::role()->get(); and $users = User::role('all')->get();
plz help on this

**Reply by jacksfero** [reply]
want to list only above two users

![image](https://user-images.githubusercontent.com/17903719/224250197-4a7d921c-3d9c-4b7e-a797-0ab59ac391f7.png)

**Reply by drbyte** [reply]
> i want to list of the all user which assigned any roles.

I assume you mean "only the users which HAVE a role assigned (and ignoring those without any roles assigned)".
That would be:
`$users_with_roles = User::has('roles')->get();`
(which is the opposite of `$users_without_any_roles = User::doesntHave('roles')->get();`) 
You can search "doesntHave" in the Laravel docs for more advanced examples of querying scopes.

**Reply by jacksfero** [reply]
Thank you so much
